### ISSUE TITLE
Block Comments

### ISSUE THREAD

**Issue by DireFolly** [mermaid-js/mermaid#1249]
**Describe the solution you'd like**
Block Commenting for verbose explanations and pasting large bodies of text would be very helpful.

Something like:

%%*
Blah blah, blah.

Blah?

Blah.
*%%

**Additional context**
My team uses mermaid & git to document and version control complex processes where there is often a need for detailed explanations.

**Comment by GDFaber** [comment]
Sorry, this is not resolved yet, I had the changes merged by accident.

Can you please create another pull request containing the changes? Please add unit tests for your changes as well. The ability to use block comments should also be included into the mermaid docs.

Sorry for the inconvenience, @erelling.

**Comment by erelling** [comment]
Thanks, no problem😄
Will do next time around, likely the week after next.

**Comment by jgreywolf** [comment]
@erelling  What is the status of this?  Still working on it?

**Comment by erelling** [comment]
Yes jgreywolf, thanks for the reminder! I'll have this done shortly.

**Comment by PatrickHonorez** [comment]
that would be nice to have! Is this issue abandonned ?

**Comment by jgreywolf** [comment]
@erelling I have unassigned you, and opened this up for contributors - please respond if you are still intending to resolve this

**Comment by natikagg** [comment]
Is this solved yet?

**Comment by nirname** [comment]
@natikagg no, it is not, contributions are welcome

**Comment by eugene27k** [comment]
This is the most wanted feature, I vote for this

**Comment by ZeroMarker** [comment]
I want this too

**Comment by Olegt0rr** [comment]
+1

**Comment by kobejeedeluna-ms** [comment]
+1

**Comment by wanglbox** [comment]
mostly voted feature +1

**Comment by StepanSaenko** [comment]
+1

**Comment by Mosaibah** [comment]
+1

**Comment by digiajay** [comment]
+1

**Comment by mdaffernaderant** [comment]
+1

**Comment by vladislavkn** [comment]
+1

**Comment by yaksetig** [comment]
+1

**Comment by bwinsley** [comment]
Hi, I'm happy to add this feature!
@GDFaber would it just be re-adding your previous commit and adding unit tests?

**Comment by GDFaber** [comment]
> Hi, I'm happy to add this feature! [@GDFaber](https://github.com/GDFaber) would it just be re-adding your previous commit and adding unit tests?

Hi @bwinsley, happy to hear you want to contribute and add this feature to Mermaid.

Unfortunately, I've been mostly inactive in this project for the past few years, so I cannot really tell if the approach taken by @erelling would still fit in. Mermaid has been widely refactored as far as I can tell, so there's a chance that the PR needs to be adapted to today's codebase and markdown syntax.

@

**Comment by bwinsley** [comment]
Sorry for the wait, it has been a busy time for me.

The PR for this is ~~ready now~~ (almost ready, sorry having some trouble with passing the CI and with linting) and has support for block comments.

- The first line must begin with %%* for the block comment to be recognised (White space before it is allowed)
- The last line must end with *%% for the block comment to be recognised
- The first line is allowed to also be the last line

**Comment by bwinsley** [comment]
Having trouble with the CI and what seems to be an issue with cache in the PR's CI. I'm closing the PR and reopening in hopes that restarting it from scratch will work. Otherwise, it would be great if a mermaid maintainer could assist with working out the issue with the CI because I can't find anything wrong on my end
